packages/suite/src/views/onboarding/steps/ResetDevice.tsx

```diff
@@ -1,111 +1,154 @@
             heading={<Translation id="TR_ONBOARDING_GENERATE_SEED" />}
-            description={<Translation id="TR_ONBOARDING_GENERATE_SEED_DESCRIPTION" />}
+            description={
+                device_model !== undefined && canChooseBackupType(device_model) ? (
+                    <Translation
+                        id="TR_ONBOARDING_SELECT_SEED_TYPE"
+                        values={{
+                            primary: chunks => <Text variant="primary">{chunks}</Text>,
+                        }}
+                    />
+                ) : (
+                    <Translation id="TR_ONBOARDING_CANNOT_SELECT_SEED_TYPE" />
+                )
+            }
             device={isWaitingForConfirmation ? device : undefined}
             isActionAbortable={isActionAbortable}
             outerActions={
-                !isWaitingForConfirmation ? (
+                !isWaitingForConfirmation && !isDeviceLocked ? (
                     // There is no point to show back button if user can't click it because confirmOnDevice bubble is active
                     <OnboardingButtonBack onClick={() => goToPreviousStep()} />
                 ) : undefined
             }
+            variant="small"
         >
-            {!isWaitingForConfirmation ? (
-                // Show options to chose from only if we are not waiting for confirmation on the device (because that means user has already chosen )
-                <OptionsWrapper $fullWidth={false}>
-                    <OnboardingOption
-                        icon="SEED_SINGLE"
-                        data-test={
-                            isShamirBackupAvailable
-                                ? '@onboarding/button-standard-backup'
-                                : '@onboarding/only-backup-option-button'
-                        }
-                        onClick={isDisabled ? undefined : handleSingleSeedReset}
-                        heading={<Translation id="SINGLE_SEED" />}
-                        description={<Translation id="SINGLE_SEED_DESCRIPTION" />}
-                    />
-
-                    {isShamirBackupAvailable && (
-                        <>
-                            <OptionsDivider />
-                            <OnboardingOption
-                                icon="SEED_SHAMIR"
-                                data-test="@onboarding/shamir-backup-option-button"
-                                onClick={isDisabled ? undefined : handleShamirReset}
-                                heading={<Translation id="SHAMIR_SEED" />}
-                                description={<Translation id="SHAMIR_SEED_DESCRIPTION" />}
-                            />
-                        </>
-                    )}
+            {!isWaitingForConfirmation && !isDeviceLocked ? (
+                <OptionsWrapper $fullWidth={true}>
+                    <SelectWrapper>
+                        {device_model !== undefined && canChooseBackupType(device_model) && (
```

Comment:
This condition is used twice - how about putting it in a variable and use it on both places for better readability

Reply:
nica catch! fixed